### DISCUSSION TITLE
Dimensionner dynamiquement un contrôle en lecture seul en fonction de sa valeur ?

### DISCUSSION THREAD

**Discussion by MJMESS** [MTESSDev/FRW#46] [Annonces]
Bonjour, 

est-ce possible de faire en sorte que la taille d'un contrôle text ou textarea en lecture seul s'ajuste automatiquement en fonction du contenu de la valeur de pré-remplissage ?

Merci,

**Comment by j0ber** [comment]
Bonjour,

Pour un contrôle texte, il n'est pas possible de modifier la taille (largeur) autrement qu'en appliquant une classe au contrôle voir [taille des contrôles](https://formulaires.it.mtess.gouv.qc.ca/Form/7/P700U/0/N/#p=tailleControle)  La largeur ne s'ajuste pas au contenu mais devrait être configurée en fonction de la largeur appropriée au contenu attendu.

C'est la même chose pour la largeur des textarea.

Dans votre cas j'imagine que pour les textarea c'est la hauteur qui vous intéresse?

Je suis curieux de voir le cas de figure d'un textarea lecture seule? Pouvez-vous me donner votre exemple? (Screenshot)
Le textarea n'est peut-être pas le contrôle approprié.

**Comment by MJMESS** [comment]
Bonjour, oui c'est la hauteur qui nous intéresse, en fait il s'agit d'une adresse prérempli que nous affichons dans un champ textarea. Dépendamment de l'informations de l'adresse parfois c'est sur 2, 3 ou 4 lignes. Aussi lorsque nous mettons /n pour les sauts de ligne, ça fonctionne bien, cependant, dans le gabarit par défaut, l'alignement n'est pas parfait. Les lignes supplémentaires s'affichent complètement à gauche et ne son pas aligné avec la première ligne. Pour vous montrer, dans l'exemple, je n'ai pas mis le champ en lecture seul. Sinon est-ce que nous pouvons fixer la hauteur du contrôle à l'avance afin qu'il ait la bonne taille pour voir l'adresse complète si jamais il y a le nombre de ligne maximum possible pour une adresse.

![image](https://user-images.githubusercontent.com/122536311/229797652-f6e87ac4-a80b-43c5-9ee0-a9fbf40cd338.png)

Merci!

**Reply by j0ber** [reply]
Concernant la hauteur (nombre de lignes) du textarea il est possible de le spécifié. Vous pourriez mettre 4 ou 5 lignes comme valeur pour votre champ et le contenu serait toujours visible dans son entièreté.

J'ai bonifié notre P700 car nous n'avions pas d'exemple avec le champ multiligne. [Regardez l'exemple 2](https://formulaires.it.mtess.gouv.qc.ca/Form/7/P700U/0/N/#p=sectionTexteMultiligne).

Pour ce qui est d'un ajustement de la hauteur automatique, ce n'est pas prévu pour le moment. Ce sera peut-être ajouté si plusieurs en font la demande.

Concernant le PDF nous allons regarder s'il est possible de générer un texte multiligne **_sous_** le libellé. Ça éviterait votre problème. Nous allons vous revenir  à ce sujet.

**Comment by MJMESS** [comment]
Super merci beaucoup!

**Comment by MJMESS** [comment]
Bonjour, dans l'outil FRW - Démarrage d'essais ça fonctionne bien mais une fois déployé il ne fait pas les sauts de ligne et affiche les /n. Devons nous faire quelque chose de particulier pour avoir le même résultat ou s'agit-il d'un bug ?

![image](https://user-images.githubusercontent.com/122536311/230175494-1494a214-3c43-40b6-b4c6-82d899aa959c.png)

Merci

**Comment by j0ber** [comment]
C'est probablement votre traitement. Avez-vous "escapé" les backslash?  Ça semble le cas.

**Comment by MJMESS** [comment]
Ok, je m'attendais à pouvoir prendre la valeur de pré-remplissage qui fonctionne dans l'outil de développement afin de l'envoyer à votre api et obtenir le même résultat.
 
Donc dans l'outil de développement nous devons mettre ceci : "123, rue de l'Organisme\nTrois-Rivières (Québec)\nG1G 1G1"
Mais lorsque nous appelons votre api, de ce que je comprend nous devons doubler les backslash afin d'obtenir le même résultat.

C'est bien ça ?

**Reply by MJMESS** [reply]
J'ai essayé et ça ne fonctionne pas, je n'ai pas bien compris ce que je dois vous envoyer pour que cela fonctionne. J'ai debbugé afin de m'assurer de ce que je vous envoi. Et si je vous envois : "123, rue de l'Organisme\nTrois-Rivières (Québec)\nG1G 1G1" je vois que les backslash sont doublés de votre côté dans le contenu du modèle de données. Donc lors de mon essai pour ajouter un backslash supplémentaire, ça a juste encore doublés donc ça devient 4 backslash. Et habituellement nous ajoutons un backslash si nous voulons que ce soit interprété comme du texte je crois.

Est-ce que vous avez un exemple sur la façon dont nous devons vous envoyer l'information lorsqu'il y a des sauts de lignes ?

Merci

**Comment by j0ber** [comment]
Créez une issue à partir de la discussion. (lien "Create issue from discussion" )
Nous allons vérifier que c'est bien de notre côté et faire un correctif si c'est le cas.
